autorest/azure/environments.go

```diff
@@ -0,0 +1,99 @@
+package azure
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+)
+
+// Environment represents a set of endpoints for each of Azure's Clouds.
+type Environment struct {
+	Name                      string
+	ManagementPortalURL       string
```

Comment:
I highly doubt we need ManagementPortalURL and PublishSettingsURL. they're browser endpoints. OTOH `StorageEndpointSuffix` now has duplication with `azure-sdk-for-go/storage` package, which actually doesn't use autorest, but assuming other people may use it, so we can keep that one.